liquibase-service/src/main/resources/db/rdb_modern/routines/005-sp_nrt_investigation_postprocessing-003.sql

```diff
@@ -146,6 +159,196 @@ BEGIN
+                    max(CASE WHEN t.cd = 'INV155' THEN t.response
+                             ELSE NULL END) AS IMPORT_FRM_CITY,
+                    max(CASE WHEN d.cd = 'INV132' THEN d.response
+                             ELSE NULL END) AS HSPTL_ADMISSION_DT,
+                    max(CASE WHEN d.cd = 'INV133' THEN d.response
+                             ELSE NULL END) AS HSPTL_DISCHARGE_DT,
+                    max(CASE WHEN n.cd = 'INV134' THEN n.response
+                             ELSE NULL END) AS HSPTL_DURATION_DAYS
+                INTO #final_inv_obs
+                FROM #temp_inv_obs tnio
+                         LEFT JOIN #tmp_coded c on tnio.public_health_case_uid = c.public_health_case_uid
+                         LEFT JOIN #tmp_txt t on tnio.public_health_case_uid = t.public_health_case_uid
+                         LEFT JOIN #tmp_num n on tnio.public_health_case_uid = n.public_health_case_uid
+                         LEFT JOIN #tmp_date d on tnio.observation_id = d.observation_id
+                GROUP BY tnio.public_health_case_uid;
+
+                IF @debug = 'true' SELECT '#final_inv_obs', * FROM #final_inv_obs;
+
+                /*Update statement considers Investigation codes that are mapped directly. The non-null condition is selected.*/
+                UPDATE tmp
+                SET
+                    HSPTLIZD_IND = COALESCE(tmp.HSPTLIZD_IND, obs.HSPTLIZD_IND),
+                    IMPORT_FRM_CNTRY = COALESCE(tmp.IMPORT_FRM_CNTRY,obs.IMPORT_FRM_CNTRY),
+                    IMPORT_FRM_STATE_CD = COALESCE(tmp.IMPORT_FRM_STATE_CD, obs.IMPORT_FRM_STATE_CD),
+                    IMPORT_FRM_CNTY_CD = COALESCE(tmp.IMPORT_FRM_CNTY_CD, obs.IMPORT_FRM_CNTY_CD),
+                    FOOD_HANDLR_IND = COALESCE(tmp.FOOD_HANDLR_IND, obs.FOOD_HANDLR_IND),
+                    PATIENT_PREGNANT_IND = COALESCE(tmp.PATIENT_PREGNANT_IND, obs.PATIENT_PREGNANT_IND),
+                    DAYCARE_ASSOCIATION_IND = COALESCE(tmp.DAYCARE_ASSOCIATION_IND, obs.DAYCARE_ASSOCIATION_IND),
+                    DIE_FRM_THIS_ILLNESS_IND = CASE WHEN obs.DIE_FRM_THIS_ILLNESS_IND_INV145 IS NOT NULL THEN obs.DIE_FRM_THIS_ILLNESS_IND_INV145
+                                                    ELSE COALESCE(tmp.DIE_FRM_THIS_ILLNESS_IND,obs.DIE_FRM_THIS_ILLNESS_IND_PRT103, obs.DIE_FRM_THIS_ILLNESS_IND_MEA078, obs.DIE_FRM_THIS_ILLNESS_IND_RUB162)
+                        END,
+                    IMPORT_FRM_CITY = COALESCE(tmp.IMPORT_FRM_CITY, obs.IMPORT_FRM_CITY),
+                    HSPTL_ADMISSION_DT = COALESCE(tmp.HSPTL_ADMISSION_DT, obs.HSPTL_ADMISSION_DT),
+                    HSPTL_DISCHARGE_DT = COALESCE(tmp.HSPTL_DISCHARGE_DT, obs.HSPTL_DISCHARGE_DT),
+                    HSPTL_DURATION_DAYS = COALESCE(tmp.HSPTL_DURATION_DAYS, obs.HSPTL_DURATION_DAYS)
+                FROM #temp_inv_table tmp
+                         LEFT JOIN #final_inv_obs obs on tmp.case_uid = obs.public_health_case_uid;
```

Comment:
👍 I was checking to see if this could be refactored into #temp_inv_table's definition, but it seems like it is better to leave this as an update statement so you can make #temp_inv_table first and you only have to filter one time and use it to make all of the subsequent temp tables. Was this also your reasoning?

Reply:
Yes. This has me thinking of an alternative approach now- _left join_ instead of _update with COALESCE_. Functionally what this needs update the null values or leave the value as is- a left join with case statement should be able to do that as well.

Reply:
I think you have a pretty good case for leaving this as an update statement, but I am interested to know what's faster between the two options. Worth mentioning that even if you do make a new temp table, you can still use COALESCE instead of a case statement where applicable. Would be a little more succinct.